### PR TITLE
IF-278: refactoring publication state

### PR DESCRIPTION
**Pull request by fkn-iteratec**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-278 Refactoring für Publish-State von AssetTypeTemplates](https://app.clickup.com/t/2177764/IF-278)

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/PublicationState.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.fusion.fusionbackend.model.enums;
+
+public enum PublicationState {
+
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/PublicationState.java:19]: consider removing empty line

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetTypeTemplateService.java [hunk 2]
```diff
@@ -65,22 +67,41 @@ public AssetTypeTemplate getAssetTypeTemplate(final Long assetTypeTemplateId, fi
...
         final AssetType assetType = assetTypeService.getAssetType(assetTypeId);
+
+        validate(assetTypeTemplate, assetType);
+
+        assetTypeTemplate.setAssetType(assetType);
+
+        return assetTypeTemplateRepository.save(assetTypeTemplate);
+    }
+
+    private void validate(AssetTypeTemplate assetTypeTemplate, AssetType assetType) {
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetTypeTemplateService.java:78]: consider a more detailed name like e.g. "validateAssetTypeTemplateWithAssetType"

File: fusionfrontend/src/app/store/asset-type-template/asset-type-template.model.ts [hunk 3]
```diff
@@ -26,9 +26,14 @@ export class AssetTypeTemplate extends BaseEntity {
...
   assetType: AssetType;
-  published: boolean;
+  publicationState: PublicationState;
   publishedDate: Date;
   publishedVersion: bigint;
   fieldTargetIds: Array<ID>;
   fieldTargets: Array<FieldTarget>;
 }
+
+export enum PublicationState {
```
Comment (jsylupp) [inline fusionfrontend/src/app/store/asset-type-template/asset-type-template.model.ts:36]: enum also exists in backend. Version in backend is without value assignment. Is this difference on purpose?

**Review by jsylupp — CHANGES_REQUESTED** [review]

**Pull request merged**